src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -644,41 +644,42 @@ Transactor::apply()
 }
 
 NotTEC
-Transactor::checkSign(PreclaimContext const& ctx)
+Transactor::checkSign(
+    PreclaimContext const& ctx,
+    AccountID const& idAccount,
+    STObject const& sigObject)
 {
-    auto const pkSigner = ctx.tx.getSigningPubKey();
+    auto const pkSigner = sigObject.getFieldVL(sfSigningPubKey);
     // Ignore signature check on batch inner transactions
-    if (ctx.tx.isFlag(tfInnerBatchTxn) &&
+    if (sigObject.isFlag(tfInnerBatchTxn) &&
         ctx.view.rules().enabled(featureBatch))
     {
         // Defensive Check: These values are also checked in Batch::preflight
-        if (ctx.tx.isFieldPresent(sfTxnSignature) || !pkSigner.empty() ||
-            ctx.tx.isFieldPresent(sfSigners))
+        if (sigObject.isFieldPresent(sfTxnSignature) || !pkSigner.empty() ||
+            sigObject.isFieldPresent(sfSigners))
         {
             return temINVALID_FLAG;  // LCOV_EXCL_LINE
         }
         return tesSUCCESS;
     }
 
     if ((ctx.flags & tapDRY_RUN) && pkSigner.empty() &&
-        !ctx.tx.isFieldPresent(sfSigners))
+        !sigObject.isFieldPresent(sfSigners))
     {
         // simulate: skip signature validation when neither SigningPubKey nor
         // Signers are provided
         return tesSUCCESS;
     }
 
-    auto const idAccount = ctx.tx[~sfDelegate].value_or(ctx.tx[sfAccount]);
-
     // If the pk is empty and not simulate or simulate and signers,
     // then we must be multi-signing.
     if (ctx.tx.isFieldPresent(sfSigners))
     {
-        STArray const& txSigners(ctx.tx.getFieldArray(sfSigners));
-        return checkMultiSign(ctx.view, idAccount, txSigners, ctx.flags, ctx.j);
+        return checkMultiSign(ctx, idAccount, sigObject);
     }
 
     // Check Single Sign
+    // This ternary is only needed to handle `simulate`
```

Comment:
can you change the logging / assert below  to say `checkSign` ? No need to show overload details, since the other overload doesn't do assertions or logging

Also, there is no ternary here anymore, you can shorten the assert description.

Reply:
> can you change the logging / assert below to say `checkSign` ? No need to show overload details, since the other overload doesn't do assertions or logging
> 
> Also, there is no ternary here anymore, you can shorten the assert description.

Fixed